.github/workflows/build.yml

```diff
@@ -5,6 +5,12 @@ on: [push, pull_request]
 # The below variables reduce repetitions across similar targets
 env:
   REMOVE_BUNDLED_BOOST : rm -rf /usr/local/share/boost
+  BUILD_DEFAULT_LINUX: |
```

Comment:
Can be removed again.

Reply:
No problem, if you're fine with the repetition, that this removal will induce.

Reply:
Do you mean the ccache line? I don't see much repetition otherwise.

Reply:
I mean line 92 and line 152.

Reply:
Ignore my other comments and add

```
ccache --max-size=150M
cmake -S . -B build -D ARCH="default" -D BUILD_TESTS=ON -D CMAKE_BUILD_TYPE=release && cmake --build build -j3
```
here and `cmake --build build --target test` to line 153

Reply:
This won't be enough for the 153, since ARCH="default" isn't selected implicitly, therefore: 
- the build job's cache will be read but not used
- the cache with such native builds from other machines will be incompatible sooner or later

Reply:
"Ignore my other comments", meant that you should keep line 93 as `      run: ${{env.BUILD_DEFAULT_LINUX}}`

Reply:
Ah cool.

Reply:
It seems to work great.